### DISCUSSION TITLE
Can I make a custom operator with function as one of inputs?

### DISCUSSION THREAD

**Discussion by DenisSvirin** [MilesCranmer/PySR#231] [Q&A]
Is it possible to create function like this: function_1(function_2, variable_1), so I can apply function_2 to variable_1 inside function_1? 
Preferebly without drastic backend changes

**Comment by MilesCranmer** [comment]
Not sure I understand, sorry - why can’t you just make a new function equal to the composition?

**Reply by DenisSvirin** [reply]
Because i dont know functions.
I want regression to find one.
Like if i have F(cos, x), i would like to use cos() function inside F, not the result cos(x)

**Reply by MilesCranmer** [reply]
In this example, what would `F` do to `cos`?

**Reply by DenisSvirin** [reply]
f(cos, variable_1):
return cos(10 + 4 * variable_1) + cos(30 + 2 * variable_1)

So in general f is like: function_2(10 + 4 * variable_1) + function_2(30 + 2 * variable_1)

**Reply by MilesCranmer** [reply]
Ah I see what you mean. So f is a function on a function.

Unfortunately this isn’t possible right now; all functions must be defined as f: ℝ × ℝ → ℝ.  In the future this will be extended to tensor operations. However, I think functions on functions is really difficult to set up, unfortunately.

**Reply by MilesCranmer** [reply]
Maybe in the future I’ll make it so arbitrary datatypes are supported, and you could pass `cos` as a column of your dataset. But if `cos` is an operator, it will never show up in a leaf of an expression tree by itself without an argument.

**Reply by DenisSvirin** [reply]
Oh, ok.
Thanks
